### PR TITLE
Target Microsoft.Extensions, Microsoft.AspNetCore v2.1

### LINKED ISSUE #443: Target Microsoft.Extensions and Microsoft.AspNetCore in v2.1 (LTS)
Created as a follow-up issue from #390.

Since .NET Core / ASP .NET Core v2.1 is an LTS release most business apps will probably depend on v2.1 instead of v2.2. Currently AppMetrics targets v2.2 which forces all users to update to v2.2.

Furthermore AppMetrics seems to not use any v2.2 specific features so that a down-grade should be possible.

### PR DESCRIPTION
**Pull request by prochnowc**
### The issue or feature being addressed

Fixes #443 (as dicusses in #390).

### Details on the issue fix or feature implementation

- Modified references to target `Microsoft.Extensions.*` and `Microsoft.AspNetCore.*` in version 2.1
- Modified unit tests and benchmarks to multi-target netcoreapp2.1, netcoreapp2.2 and net461
- Left sandboxes targeting latest version (2.2)
- Cleaned up some ambiguous references (with and/or without version attribute)

### Confirm the following

- [X] I have ensured that I have merged the latest changes from the dev branch
- [X] I have successfully run a [local build](https://github.com/AppMetrics/AppMetrics#how-to-build)
- [X] I have included unit tests for the issue/feature
- [X] I have included the github issue number in my commits

### CODE REVIEW

**Comment by prochnowc** [comment]
@alhardy Glad to contribute some small changes .. please have a look at the PR.

File: src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj [hunk 1]
```diff
@@ -24,8 +24,8 @@
     <PackageReference Include="App.Metrics.Health.Checks.Process" />
     <PackageReference Include="App.Metrics.Health.Reporting.Metrics" />
     <PackageReference Include="App.Metrics.Health.Reporting.Slack" />
-    <PackageReference Include="Microsoft.AspNetCore" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" />
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
     <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="2.2.0" />
```
Comment (plachor) [inline src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj]: I think this one above is also target for this PR. It's referenced in more than one place.

File: src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj [hunk 2]
```diff
@@ -24,8 +24,8 @@
     <PackageReference Include="App.Metrics.Health.Checks.Process" />
     <PackageReference Include="App.Metrics.Health.Reporting.Metrics" />
     <PackageReference Include="App.Metrics.Health.Reporting.Slack" />
-    <PackageReference Include="Microsoft.AspNetCore" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" />
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
```
Comment (plachor) [inline src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj]: Why 2.2.0?

File: src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
   <ItemGroup>
     <PackageReference Include="App.Metrics.Core" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
```
Comment (plachor) [inline src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj]: Why no version?

File: src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj [hunk 4]
```diff
@@ -8,7 +8,7 @@
...
     <PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions" Version="2.2.0" />
```
Comment (plachor) [inline src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj]: Looks like target too

File: src/Extensions/src/App.Metrics.Health.Extensions.DependencyInjection/App.Metrics.Health.Extensions.DependencyInjection.csproj [hunk 5]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="App.Metrics.Health" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="2.2.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="2.1.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
```
Comment (plachor) [inline src/Extensions/src/App.Metrics.Health.Extensions.DependencyInjection/App.Metrics.Health.Extensions.DependencyInjection.csproj]: What behavior (outcome) you expect without version here?

Comment (plachor) [reply src/Extensions/src/App.Metrics.Health.Extensions.DependencyInjection/App.Metrics.Health.Extensions.DependencyInjection.csproj]: And one bellow

File: src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj (hunk 1 above)
Comment (prochnowc) [reply src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj]: I left the Sandboxes targetting 2.2 by intention. As I undestood this is playground for new features, so it doesnt really matter.

File: src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj (hunk 2 above)
Comment (prochnowc) [reply src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj]: Same about Sandboxes here.

File: src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj (hunk 3 above)
Comment (prochnowc) [reply src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj]: Please check `Directory.Build.targets` in the top-level directory. Versions are globally updated there.

File: src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj (hunk 4 above)
Comment (prochnowc) [reply src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj]: If you mean reference to `Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions` .. it does not exist in 2.1 - we have to reference 2.2 here.

File: src/Extensions/src/App.Metrics.Health.Extensions.DependencyInjection/App.Metrics.Health.Extensions.DependencyInjection.csproj (hunk 5 above)
Comment (prochnowc) [reply src/Extensions/src/App.Metrics.Health.Extensions.DependencyInjection/App.Metrics.Health.Extensions.DependencyInjection.csproj]: Same here - versions updated globally in top-level `Directory.Build.targets`

File: src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj (hunk 1 above)
Comment (plachor) [reply src/AspNetCoreHealth/sandbox/HealthSandboxMvc/HealthSandboxMvc.csproj]: ok :)

File: src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj (hunk 3 above)
Comment (plachor) [reply src/Extensions/src/App.Metrics.Extensions.HealthChecks/App.Metrics.Extensions.HealthChecks.csproj]: Great thnx for clarification

**Pull request merged**